File: tests/python/cases/case_060_enrich_ledger.py
"""
Task case 060: enrich ledger payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_060_enrich_ledger(input_data):
    # Task: enrich ledger records and prepare transport-ready payload.
    task_name = 'enrich_ledger'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ledger',
        'stage': 'enrich',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']

File: tests/nodejs/cases/case_060_enrich_ledger.js
/**
 * Task case 060: enrich_ledger
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case060EnrichLedger(input) {
  // Task: enrich ledger records and prepare transport-ready payload.
  const taskName = 'enrich_ledger';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ledger',
    stage: 'enrich',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case060EnrichLedger,
};
